src/Workspaces/Core/Portable/Utilities/EditDistance.cs

```diff
@@ -186,24 +517,46 @@ public static bool IsCloseMatch(string originalText, string candidateText)
         /// </summary>
         public static bool IsCloseMatch(string originalText, string candidateText, out double matchCost)
         {
-            return IsCloseMatch(originalText, candidateText, GetCloseMatchThreshold(originalText), out matchCost);
+            using (var editDistance = new EditDistance(originalText))
+            {
+                return editDistance.IsCloseMatch(candidateText, out matchCost);
+            }
         }
 
-        public static int GetCloseMatchThreshold(string originalText)
+        public bool IsCloseMatch(string candidateText, out double matchCost)
         {
-            return Math.Min(4, originalText.Length / 2);
+            if (this.originalText.Length < 3)
+            {
+                // If we're comparing strings that are too short, we'll find 
+                // far too many spurious hits.  Don't even both in this case.
+                matchCost = double.MaxValue;
+                return false;
```

Comment:
What about exact matches? Maybe those are filtered out earlier, but that's not clear here.